src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -149,8 +157,9 @@ public static int fuzzyLowCount(int rollupCount) {
     return rollupCount == 0 ? rollupCount : Math.max(rollupCount, 19);
   }
 
-  public EnumerateSnapshotAccessRequest enumerateByDatasetId(UUID id) {
-    return convertToEnumerateModel(snapshotRequestDao.enumerateByDatasetId_old(id));
+  public EnumerateSnapshotAccessRequest enumerateSnapshotRequests(Set<UUID> authorizedResources) {
```

Comment:
TDR doesn't do this much, but I think it's better to use a more general type for collections, unless the type is required. In this case, this method (and the methods it calls) would work with either `Set<UUID>` or `Collection<UUID>`, so I would use `Collection` here. That way if at some point we end up calling this with `List<UUID>` we don't need to change this code, or convert the `List` to a `Set` in the caller.
```suggestion
  public EnumerateSnapshotAccessRequest enumerateSnapshotRequests(Collection<UUID> authorizedResources) {
```